ee/elf-loader/src/elf.c

```diff
@@ -0,0 +1,83 @@
+/*
+# _____     ___ ____     ___ ____
+#  ____|   |    ____|   |        | |____|
+# |     ___|   |____ ___|    ____| |    \    PS2DEV Open Source Project.
+#-----------------------------------------------------------------------
+# (c) 20020 Francisco Javier Trujillo Mata <fjtrujy@gmail.com>
+# Licenced under Academic Free License version 2.0
+# Review ps2sdk README & LICENSE files for further details.
+*/
+
+#include <string.h>
+#include <sifrpc.h>
+#include <kernel.h>
+#include <sys/stat.h>
+#include <stdbool.h>
+#include <malloc.h>
+
+#include "elf.h"
+
+// Loader ELF variables
+extern u8 loader_elf[];
+extern int size_loader_elf;
+
+// ELF-loading stuff
+#define ELF_MAGIC 0x464c457f
+#define ELF_PT_LOAD 1
+
+int printf(const char *format, ...);
+
+static bool file_exists(const char *filename) {
+  struct stat   buffer;   
+  return (stat (filename, &buffer) == 0);
+}
+
+int LoadELFFromFile(const char *filename, int argc, char *argv[])
+{
+	u8 *boot_elf;
+	elf_header_t *eh;
+	elf_pheader_t *eph;
+	void *pdata;
+	int i;
+	int new_argc = argc + 1;
+	char **new_argv = malloc(new_argc);
+	
+	// We need to check that the ELF file before continue
+	if (!file_exists(filename)) {
+		return -1; // ELF file doesn't exists
```

Comment:
If the file doesn't exist, you're leaking the memory just allocated for `new_argv`

Reply:
Thanks!